### PR TITLE
Add support for the anonymous user account and record python dependencies for testing

### PR DESCRIPTION
**Pull request by korydraughn**
All tests pass with this change.

Also, the server is now detecting the absence of the anonymous user and reporting the appropriate error. I think the original behavior I was seeing was due to old build artifacts. Recompiling from scratch resolved the issue.

### CODE REVIEW

**Comment by trel** [comment]
oh, well that's the best answer...

File: API.md [hunk 1]
```diff
@@ -8,13 +8,15 @@ If you discover that some topic related to the behavior of the endpoint operatio
 
 ### Scheme: Basic
 
-_Command:_
+#### Request
+
 ```bash
-curl http://localhost:<port>/irods-http-api/<version>/authenticate \
-    -X POST --user <username>:<password> 
+curl -X POST -u <username>:<password> http://localhost:<port>/irods-http-api/<version>/authenticate
```
Comment (trel) [inline API.md:14]: just confirming...  password is plaintext here?

Comment (korydraughn) [reply API.md:14]: Yes.

If `-u <username>` is used instead, curl will prompt for the password without echoing the input. Mentioning that is out of scope IMO.

Comment (trel) [reply API.md:14]: got it.   carry on.

**Comment by trel** [comment]
looks good - appears ready for squash/pound to me.

**Comment by korydraughn** [comment]
#'d

**Comment by trel** [comment]
mergin'.

**Pull request merged**